src/registrar/utility/csv_export.py

```diff
@@ -161,19 +162,20 @@ def write_csv(
     # Reduce the memory overhead when performing the write operation
     paginator = Paginator(all_domain_infos, 1000)
 
-    if get_domain_managers and len(all_domain_infos) > 0:
-        # We want to get the max amont of domain managers an
-        # account has to set the column header dynamically
-        max_dm_count = max(len(domain_info.domain.permissions.all()) for domain_info in all_domain_infos)
-        update_columns_with_domain_managers(columns, max_dm_count)
-
-    if should_write_header:
-        write_header(writer, columns)
+    # The maximum amount of domain managers an account has
+    # We get the max so we can set the column header accurately
+    max_dm_count = 0
+    total_body_rows = []
 
     for page_num in paginator.page_range:
         rows = []
         page = paginator.page(page_num)
         for domain_info in page.object_list:
+            # Get count of all the domain managers for an account
+            dm_count = domain_info.domain.permissions.count()
```

Comment:
**Question:** Just to confirm -- because we're not doing the domain_info call outside, and bc we already have it and it's within the loop that helps the time intensiveness, correct?

Reply:
That's the hope

Reply:
I think I made a very similiar suggestion to zander's here, but on the actual loop in question. and I think moving it here would save on having a bool passed to the other function.